src/test/bridges/convex/ConvexStakingBridge.t.sol

```diff
@@ -0,0 +1,518 @@
+
+        vm.prank(invalidCaller);
+        vm.expectRevert(ErrorLib.InvalidCaller.selector);
+        bridge.convert(
+            AztecTypes.AztecAsset(1, curveLpToken, AztecTypes.AztecAssetType.ERC20),
+            emptyAsset,
+            AztecTypes.AztecAsset(100, rctClone, AztecTypes.AztecAssetType.ERC20),
+            emptyAsset,
+            depositAmount,
+            0,
+            0,
+            BENEFICIARY
+        );
+    }
+
+    // pool not loaded yet, RCT not deployed yet
+    function testPoolNotLoadedYet(uint96 _depositAmount) public {
+        vm.assume(_depositAmount > 1);
+
+        _setupBridge(112);
+        _setupRepresentingConvexTokenClone();
+
+        // Mock initial balance of CURVE LP Token for Rollup Processor
+        deal(curveLpToken, rollupProcessor, _depositAmount); // could be removed
+        // transfer CURVE LP Tokens from RollUpProcessor to the bridge
+        IERC20(curveLpToken).transfer(address(bridge), _depositAmount); // could be removed
+
+        vm.expectRevert(bytes(""));
+        bridge.convert(
+            AztecTypes.AztecAsset(1, curveLpToken, AztecTypes.AztecAssetType.ERC20),
+            emptyAsset,
+            AztecTypes.AztecAsset(100, rctClone, AztecTypes.AztecAssetType.ERC20),
+            emptyAsset,
+            _depositAmount,
+            0,
+            0,
+            BENEFICIARY
+        );
+    }
+
+    function testStakeLpTokens(
+        uint96 _depositAmount,
+        uint16 _poolId1,
+        uint16 _poolId2,
+        uint16 _poolId3,
+        uint16 _poolId4,
+        uint16 _poolId5
+    ) public {
+        vm.assume(_depositAmount > 1);
+
+        uint256 poolLength = IConvexBooster(BOOSTER).poolLength();
+
+        _setupTestPoolIds(poolLength, _poolId1, _poolId2, _poolId3, _poolId4, _poolId5);
+
+        for (uint256 i = 0; i < poolIdsToTest.length; i++) {
+            if (_skipPool(i)) {
+                continue;
+            }
+
+            _setupBridge(poolIdsToTest[i]);
+            _loadPool(poolIdsToTest[i]);
+            _setupRepresentingConvexTokenClone();
+            _setupSubsidy();
+
+            skip(1 days);
+            _deposit(_depositAmount);
+            rewind(1 days);
+
+            SUBSIDY.withdraw(BENEFICIARY);
+            assertGt(BENEFICIARY.balance, 0, "Subsidy was not claimed");
+        }
+    }
+
+    function testWithdrawLpTokens(
+        uint64 _withdrawalAmount,
+        uint16 _poolId1,
+        uint16 _poolId2,
+        uint16 _poolId3,
+        uint16 _poolId4
+    ) public {
+        vm.assume(_withdrawalAmount > 4012207);
+
+        uint256 poolLength = IConvexBooster(BOOSTER).poolLength();
+
+        bool testedPoolOpen;
+
+        _setupTestPoolIds(poolLength, _poolId1, _poolId2, _poolId3, _poolId4, 4); // Pool id 4 intentionally selected -> to prevent possibility that all selected pools yield zero CRV and CVX rewards in the specified time frame.
+
+        for (uint256 i = 0; i < poolIdsToTest.length; i++) {
```

Comment:
Why are you iterating through all the pools and then skipping some? I would simply have 1 poolId as an input from the fuzzer and test it on that 1 pool. Default configuration of Foundry is running a 256 fuzz runs on each test runs so testing just 1 pool here should be ok.

Reply:
I do not iterate over all the pools, however, I selected 5 random pools from all the pools and these 5 pool Ids are then saved in poolIdsToTest and then it runs test for all five selected poolIds. The skipping mechanism is necessary to prevent tests to fails. (some pools just fail because they are not fully implemented, some pools can be close and that would fail the test, same pool cannot be tested within the same test twice, sometimes I need to add my own pool to make sure the tested assertion passes - e.g. here the pool id 4)
Do you mean that each fuzz parameters is changed 256 within the test? I read that somewhere but it didn't look like it from the test log (-vvvv)
Should I then use only a single fuzz poolId instead of five of them?

Reply:
I am not sure if this is a good concept. I will essentially need to limit the generated poolId in order to bound it to the possible poolId range and check if the pool will not fail. Run the check in a while loop to guarantee that the selected pool will really pass the checks. Also the generated poolId has to be always greater than the number of available pools (vm.assume) otherwise bound could get stuck on the single failing poolId (last time I checked the bound function, if the number fit in the given range, the number was accepted and not changed anymore.). I sure can do all this and make sure that _getPoolId() will always return a valid pool but I think this is a lot of work when instead I could just skip some of the 256 tested pools.
Also for the withdraw test to check whether the rewards were claimed, a single poolId will not be enough, I will need the poolId 4 as well.
I'll brain a little more about this but today I didn't find any better solution for the _getPoolId()